src/main/java/today/sesac/versebyverse/report/entity/Report.java

```diff
@@ -0,0 +1,162 @@
+
+    /**
+     * 신고의 고유 식별자입니다.
+     */
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    /**
+     * 신고를 한 회원입니다.
+     */
+    @ManyToOne
+    @JoinColumn(name = "reporter_id", nullable = false)
+    private Member reporter;
+
+    /**
+     * 신고된 게시글입니다. 댓글 신고 시에는 null입니다.
+     */
+    @ManyToOne
+    @JoinColumn(name = "post_id", nullable = true)
+    private Post post;
+
+    /**
+     * 신고된 댓글입니다. 게시글 신고 시에는 null입니다.
+     */
+    @ManyToOne
+    @JoinColumn(name = "comment_id", nullable = true)
+    private Comment comment;
+
+    /**
+     * 신고 유형입니다. 게시글 또는 댓글 중 하나입니다.
+     */
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private ReportType reportType;
+
+    /**
+     * 신고 사유입니다. 스팸, 부적절한 내용 등이 포함됩니다.
+     */
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private ReasonType reasonType;
+
+    /**
+     * 신고 상세 사유입니다. 최대 500자까지 입력 가능합니다.
+     */
+    @Column(length = 500)
+    private String reasonDetail;
+
+    /**
+     * 신고 처리 상태입니다. 기본값은 PENDING(대기중)입니다.
+     */
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private StatusType statusType = StatusType.PENDING;
+
+    /**
+     * 신고를 생성하는 생성자입니다.
+     *
+     * @param reporter     신고를 한 회원
+     * @param post         신고된 게시글 (댓글 신고 시 null)
+     * @param comment      신고된 댓글 (게시글 신고 시 null)
+     * @param reportType   신고 유형
+     * @param reasonType   신고 사유
+     * @param reasonDetail 신고 상세 사유
+     */
+    private Report(Member reporter, Post post, Comment comment,
+            ReportType reportType, ReasonType reasonType, String reasonDetail) {
+
+        this.reporter = reporter;
+        this.post = post;
+        this.comment = comment;
+        this.reportType = reportType;
+        this.reasonType = reasonType;
+        this.reasonDetail = reasonDetail;
+    }
+
+    /**
+     * 게시글 신고를 생성하는 정적 팩토리 메서드입니다.
+     *
+     * @param reporter     신고를 한 회원
+     * @param post         신고된 게시글
+     * @param reasonType   신고 사유
+     * @param reasonDetail 신고 상세 사유
+     * @return 생성된 Report 엔티티
+     */
+    public static Report createPostReport(Member reporter, Post post,
+            ReasonType reasonType, String reasonDetail) {
+
+        return new Report(reporter, post, null, ReportType.POST, reasonType, reasonDetail);
+    }
+
+    /**
+     * 댓글 신고를 생성하는 정적 팩토리 메서드입니다.
+     *
+     * @param reporter     신고를 한 회원
+     * @param comment      신고된 댓글
+     * @param reasonType   신고 사유
+     * @param reasonDetail 신고 상세 사유
+     * @return 생성된 Report 엔티티
+     */
+    public static Report createCommentReport(Member reporter, Comment comment,
+            ReasonType reasonType, String reasonDetail) {
+
+        return new Report(reporter, null, comment, ReportType.COMMENT, reasonType, reasonDetail);
+    }
+
+    /**
+     * 신고 상태를 승인으로 변경합니다.
+     */
+    public void accept() {
+
+        this.statusType = StatusType.ACCEPTED;
+    }
```

Comment:
P2
구두로 논의한 이야기 댓글로 남깁니다.
신고에서 처리하면, 댓글/게시글 의 is_Reported 신고 필드를 false로 변경한다고 하셨는데, 이 안에서 한번에 처리하면 까먹지 않고, 한번에 처리될 것 같아서 제안드립니다.

생각하시는 로직에 따라서 reject, alreaadyDeleted도 일관성 있게 처리 부탁드립니다.

```suggestion
    public void accept() {
        // post / comment 둘 중 하나.
        this.post.신고해제();
        this.comment.신고해제();
        this.statusType = StatusType.ACCEPTED;
    }
```

Reply:
TODO로 남겨놓고 서비스로직 작성할때 참고하겠습니다~